Sources/Logging/Logging.swift

```diff
@@ -39,20 +39,41 @@ import WASILibc
 /// logger.info("Hello World!")
 /// ```
 public struct Logger {
-    @usableFromInline
-    var handler: LogHandler
+    /// A private property to hold the boxed `LogHandler`.
+    private var _handler: Box<LogHandler>
 
     /// An identifier of the creator of this `Logger`.
     public let label: String
+    
+    /// A computed property to access the `LogHandler`.
+    public var handler: LogHandler {
+        get {
+            return _handler.value
+        }
+        set {
+            if !(isKnownUniquelyReferenced(&_handler)) {
+                _handler = Box(value: newValue)
+            } else {
+                _handler.value = newValue
+            }
+        }
+    }
 
     /// The metadata provider this logger was created with.
     public var metadataProvider: Logger.MetadataProvider? {
-        return self.handler.metadataProvider
+        return handler.metadataProvider
     }
 
     internal init(label: String, _ handler: LogHandler) {
         self.label = label
-        self.handler = handler
+        self._handler = Box(value: handler)
+    }
+}
+
+private final class Box<T> {
```

Comment:
Can we nest this type inside the `struct Logger` please. Additionally can we make it non generic and rather concrete so that it holds both the handler and the label.

Reply:
+1, often also called `Storage` but that's a nit :).